make clusterctl failue

/kind bug

```
[kris@kris cluster-api-provider-aws]$ make clusterctl
dep version || go get -u github.com/golang/dep/cmd/dep
dep: version: no such command
dep is a tool for managing dependencies for Go projects

Usage: dep <command>

Commands:

  init    Initialize a new project with manifest and lock files
  status  Report the status of the project's dependencies
  ensure  Ensure a dependency is safely vendored in the project
  prune   Prune the vendor tree of unused packages

Examples:
  dep init                          set up a new project
  dep ensure                        install the project's dependencies
  dep ensure -update                update the locked versions of all dependencies
  dep ensure github.com/pkg/errors  add a dependency to the project

Use "dep help [command]" for more information about a command.
dep ensure
GOPATH=/home/kris/go go generate ./pkg/... ./cmd/...
go fmt ./pkg/... ./cmd/...
pkg/cloud/aws/services/certificates/certificates.go
go vet ./pkg/... ./cmd/...
CGO_ENABLED=0 go install -ldflags '-extldflags "-static"'  sigs.k8s.io/cluster-api-provider-aws/clusterctl
can't load package: package sigs.k8s.io/cluster-api-provider-aws/clusterctl: cannot find package "sigs.k8s.io/cluster-api-provider-aws/clusterctl" in any of:
	/usr/lib/go/src/sigs.k8s.io/cluster-api-provider-aws/clusterctl (from $GOROOT)
	/home/kris/go/src/sigs.k8s.io/cluster-api-provider-aws/clusterctl (from $GOPATH)
make: *** [Makefile:54: clusterctl] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ GOFLAGS += -ldflags '-extldflags "-static"'
 GOREBUILD :=
 GOPATH := $(shell go env GOPATH)
 
-all: test manager clusterctl clusterawsadm
+all: check-install test manager clusterctl clusterawsadm
 
 .PHONY: vendor
 
@@ -28,6 +28,9 @@ all: test manager clusterctl clusterawsadm
 vendor:
 	dep version || go get -u github.com/golang/dep/cmd/dep
 	dep ensure
+
+check-install:
+	./scripts/check-install.sh
 
 depend-update:
 	dep version || go get -u github.com/golang/dep/cmd/dep
@@ -50,15 +53,15 @@ manager: generate fmt vet
 	CGO_ENABLED=0 go install $(GOFLAGS) $(GOREBUILD) sigs.k8s.io/cluster-api-provider-aws/cmd/manager
 
 # Build clusterctl binary.
-clusterctl: generate fmt vet
+clusterctl: check-install generate fmt vet
 	CGO_ENABLED=0 go install $(GOFLAGS) $(GOREBUILD) sigs.k8s.io/cluster-api-provider-aws/clusterctl
 
 # Build clusterawsadm binary.
-clusterawsadm: vendor
+clusterawsadm: check-install vendor
 	CGO_ENABLED=0 go install $(GOFLAGS) $(GOREBUILD) sigs.k8s.io/cluster-api-provider-aws/cmd/clusterawsadm
 
 # Build cluster-api-dev-helper binary.
-cluster-api-dev-helper: vendor
+cluster-api-dev-helper: check-install vendor
 	CGO_ENABLED=0 go install $(GOFLAGS) sigs.k8s.io/cluster-api-provider-aws/hack/cluster-api-dev-helper
 
 # Run tests

--- a/scripts/check-install.sh
+++ b/scripts/check-install.sh
@@ -1,0 +1,37 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -o errexit
+set -o nounset
+set -o pipefail
+
+WORKINGDIR=$(pwd)
+EXPECTEDDIR="${GOPATH}/src/sigs.k8s.io/cluster-api-provider-aws"
+
+if [ "${WORKINGDIR}" != "${EXPECTEDDIR}" ]; then
+    echo "------------------------------------------------------------------------------------------------------------------"
+    echo "Invalid checkout directory!"
+    echo
+    echo "Expected: ${EXPECTEDDIR}"
+    echo "Actual:   ${WORKINGDIR}"
+    echo ""
+    echo "This project assumes that the repository has been checked out to \$GOPATH/src/sigs.k8s.io/cluster-api-provider-aws"
+    echo "------------------------------------------------------------------------------------------------------------------"
+    exit 1
+fi
+
+exit 0

